### PR TITLE
Compare menus

### PR DESCRIPTION
**Pull request by oliviaxjiang**
### 1. [Issue #142: Scrolling Header](https://github.com/cuappdev/eatery-blue-android/issues/142)  
✅ **Resolved** by refactoring the EateryDetailsStickerHeader and using it in CompareMenusScreen

### 2. [Issue #141: Sheet Stays Open](https://github.com/cuappdev/eatery-blue-android/issues/141)  
✅ **Fixed** by adding an additional close line to `OnCompareMenusClick`. Did not used NavController or LaunchedEffect

### 3. [Issue #140: Formatting Stuff](https://github.com/cuappdev/eatery-blue-android/issues/140)  
🟡 **Fixed except middle format issue** Figma design is rectangular borders, so idk how to fix

### 4. [Issue #138: Row Clickable](https://github.com/cuappdev/eatery-blue-android/issues/138)  
✅ **Fixed** this issue per the outlined requirements. The app behavior should now match the expected outcome described in the issue.

### 5. [Issue #139 : eatery name bottom padding](https://github.com/cuappdev/eatery-blue-android/issues/138)  
✅ **Fixed** added padding according to figma design

### 6. [Issue #137: Not Done](https://github.com/cuappdev/eatery-blue-android/issues/137)  
🚫 **Not Addressed**: IDK what filter cuts off means.

## View Model Logic Description
CompareMenusBotSheet needs to manage its own states (adding/removing selected eateries/filters) so it gets it's own viewModel (CompareMenusBotViewModel). CompareMenusScreen also needs state management, so it also gets a view model (CompareMenusViewModel). No specific viewmodel logic resides in HomeViewModel or EateryDetailsViewModel as they are all delegated to the two new view models.

## Potential places for change (Would love advice here!)
List<Int> called eateryIds currently gets sent to CompareMenusScreen and CompareMenusScreen has to call a function to pass that information into the viewmodel, may want to used sharedStateHandle and give it directly to CompareMenusViewModel? (I tried looking up on StackOverFlow on how to share states between two screens that is connected via NavController and someone said that the best practice would be sharedViewModel (?))

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/HomeViewModel.kt
Comment (zachseidner1) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/HomeViewModel.kt:1]: Why are we changing this file? All the state for `CompareMenusBottomSheet` should be stored in the `ViewModel` that corresponds to it right?

**Review by zachseidner1 — CHANGES_REQUESTED** [review]
This is a great first draft! I like your decision to give the compare menus bottom sheet a `ViewModel` and injecting it via `hiltViewModel()`. After seeing all the filtering logic it definitely makes sense. There are a bit of comments that you need to work through, anything, so much so that anything I labeled with "nit" don't feel pressure to fix. But I really appreciate the work you've put into implementing this and I hope you've learned more about Android architecture!

File: app/src/main/java/com/cornellappdev/android/eatery/ui/components/comparemenus/CompareMenusBotSheet.kt [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+import com.cornellappdev.android.eatery.ui.viewmodels.state.CompareMenusUIState
+import com.cornellappdev.android.eatery.ui.viewmodels.state.EateryApiResponse
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+@Composable
+fun CompareMenusBotSheet(
+    onDismiss: () -> Unit,
+    onCompareMenusClick: (selectedEateries : List<Eatery>) -> Unit,
+    compareMenusBotViewModel: CompareMenusBotViewModel = hiltViewModel()
```
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/components/comparemenus/CompareMenusBotSheet.kt:63]: Ohhh okay I see what we ended up doing. I like this; a separate viewmodel for the sheet.

File: app/src/main/java/com/cornellappdev/android/eatery/ui/components/comparemenus/CompareMenusBotSheet.kt [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+                Box(
+                    modifier = Modifier
+                        .background(Color.White)
+                        .fillMaxHeight(0.4f)
+                        .fillMaxWidth()
+                ) {
+                    LazyColumn {
+                        items(eateries) { eatery ->
+                            Row(
```
[the remark this answers is not in the document]
Comment (thisjustin123) [reply app/src/main/java/com/cornellappdev/android/eatery/ui/components/comparemenus/CompareMenusBotSheet.kt:123]: Ya good point! There's a lot of indentation here.

[the remark this answers is not in the document]
Comment (thisjustin123) [reply app/src/main/java/com/cornellappdev/android/eatery/ui/components/comparemenus/CompareMenusBotSheet.kt:123]: I think it's generally good to take out the header, content, and bottom portions into separate composables for readability.

Veronica discovered this feature where you just highlight a bunch of your function and hit `Extract` that pops up and it does this all for you. I don't know how accurate/good it is though, but perhaps try it out!

File: app/src/main/java/com/cornellappdev/android/eatery/ui/components/details/EateryDetailsStickyHeader.kt [hunk 3]
```diff
@@ -61,8 +61,9 @@ fun EateryDetailsStickyHeader(
     filterText: String,
     fullMenuList: MutableList<String>,
     listState: LazyListState,
-    eateryDetailViewModel: EateryDetailViewModel = hiltViewModel(),
-    onItemClick: (Int) -> Unit
+    startIndex : Int,
+    onItemClick: (Int) -> Unit,
```
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/components/details/EateryDetailsStickyHeader.kt:65]: Nice, love the removal of viewmodel.

File: app/src/main/java/com/cornellappdev/android/eatery/ui/components/settings/AppIconBottomSheet.kt [hunk 4]
```diff
@@ -75,26 +83,50 @@ fun AppIconBottomSheet(hide: () -> Unit) {
                 Icon(Icons.Default.Close, contentDescription = "Close", tint = Color.Black)
             }
         }
+        val pagerState = androidx.compose.foundation.pager.rememberPagerState()
+        androidx.compose.foundation.pager.HorizontalPager(pageCount = 2, state = pagerState, modifier = Modifier.offset(x = -8.dp)){
+            Column{
+                when(it){
```
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/components/settings/AppIconBottomSheet.kt:89]: I smell unformatted file >:)

File: app/src/main/java/com/cornellappdev/android/eatery/ui/navigation/MainTabbedNavigation.kt [hunk 5]
```diff
@@ -52,7 +54,8 @@ import com.google.accompanist.navigation.animation.rememberAnimatedNavController
 @Composable
 fun NavigationSetup(hasOnboarded: Boolean) {
     val navController = rememberAnimatedNavController()
-    val showBottomBar = rememberSaveable { mutableStateOf(false) }
+    val showBottomBar = rememberSaveable { mutableStateOf(false)
+    }
```
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/navigation/MainTabbedNavigation.kt:58]: Strange formatting!

File: app/src/main/java/com/cornellappdev/android/eatery/ui/navigation/MainTabbedNavigation.kt [hunk 6]
```diff
@@ -419,6 +428,22 @@ fun SetupNavHost(
...
+            backStackEntry.arguments?.getString("eateryIds")?.split(",")?.map { it.toInt() }?.let { eateryIds ->
+                CompareMenusScreen(
+                    eateryIds = eateryIds,
+                    onEateryClick = {
+                        FirstTimeShown.firstTimeShown = false
+                        navController.navigate("${Routes.EATERY_DETAIL.route}/${it.id}")
+                    }
+                )
+            }
+        }
```
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/navigation/MainTabbedNavigation.kt:446]: lovely stuff, this passing of IDs is exactly what we need!

File: app/src/main/java/com/cornellappdev/android/eatery/ui/screens/CompareMenusScreen.kt
[the remark this answers is not in the document]
Comment (thisjustin123) [reply app/src/main/java/com/cornellappdev/android/eatery/ui/screens/CompareMenusScreen.kt:1]: Oh god yeah let's split this up.

File: app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/CompareMenusBotViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+class CompareMenusBotViewModel @Inject constructor(
+    private val userPreferencesRepository: UserPreferencesRepository,
+    private val eateryRepository: EateryRepository,
+    private val userRepository: UserRepository
+) : ViewModel() {
+
+    private val _compareMenusUiState = MutableStateFlow(CompareMenusUIState())
+    val compareMenusUiState: StateFlow<CompareMenusUIState> = _compareMenusUiState.asStateFlow()
+
+    val eateryFlow: StateFlow<EateryApiResponse<List<Eatery>>> =
```
[the remark this answers is not in the document]
Comment (thisjustin123) [reply app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/CompareMenusBotViewModel.kt:36]: Yes I agree. Package everything you can into 1 flow for simplicity if you can; that means `compareMenusUiState` should be the only flow exposed.

We used to violate this a lot. We shouldn't do that anymore. It makes UI and flow control a lot harder.

File: app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/HomeViewModel.kt
Comment (thisjustin123) [reply app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/HomeViewModel.kt:1]: Oh yeah good point... what are all these functions for?

**Review by thisjustin123 — COMMENTED** [review]
I'll defer the approval of this PR to zach since his comments are incredibly comprehensive. But I just left some tiny additional comments.

Thanks for the review Zach. U da man.

**Review by zachseidner1 — APPROVED** [review]
W

**Pull request merged**